src/main/scala/github/gphat/censorinus/dogstatsd/Encoder.scala

```diff
@@ -7,54 +7,57 @@ import github.gphat.censorinus.{Metric,MetricEncoder}
   */
 object Encoder extends MetricEncoder {
 
-  def encode(metric: Metric): String = metric.metricType match {
-    case "c" => encodeCounter(metric)
-    case "g" => encodeGauge(metric)
-    case "h" => encodeHistogram(metric)
-    case "m" => encodeMeter(metric)
-    case "ms" => encodeTimer(metric)
-    case "s" => encodeSet(metric)
-    case _ => "" // TODO Complain!
-  }
-
-  def encodeCounter(metric: Metric): String = {
-    s"${metric.name}:${metric.value}|c${encodeSampleRate(metric)}${encodeTags(metric)}"
-  }
-
-  def encodeGauge(metric: Metric): String = {
-    s"${metric.name}:${metric.value}|g${encodeTags(metric)}"
-  }
-
-  def encodeHistogram(metric: Metric): String = {
-    s"${metric.name}:${metric.value}|h${encodeTags(metric)}"
-  }
-
-  def encodeMeter(metric: Metric): String = {
-    s"${metric.name}:${metric.value}|m${encodeTags(metric)}"
-  }
-
-  def encodeSampleRate(metric: Metric): String = {
-    if(metric.sampleRate == 1.0) {
-      ""
-    } else {
-      s"|@${"%.4f".format(metric.sampleRate)}"
+  def encode(metric: Metric): Option[String] = metric.metricType match {
+    case "c" =>
+      val sb = new StringBuilder()
+      encodeBaseMetric(sb, metric)
+      encodeSampleRate(sb, metric.sampleRate)
+      encodeTags(sb, metric.tags)
+      Some(sb.toString())
+
+    case "g" | "h" | "m" | "ms" | "s" =>
```

Comment:
This code path should `encodeTags` in it. Probably an accident from trying to exclude the sample rate. Seems odd that the tests pass with that…